### PR TITLE
Deprecate AADAppRoleStatelessAuthenticationFilter and AADAuthenticationFilter

### PR DESCRIPTION
**Pull request by ZhuXiaoBing-cn**
* Deprecate AADAppRoleStatelessAuthenticationFilter and AADAuthenticationFilter
* Update azure-spring-boot-starter-active-directory readme file

### CODE REVIEW

**Comment by ZhuXiaoBing-cn** [comment]
For suggestions, please visit this #17860.

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAppRoleStatelessAuthenticationFilter.java [hunk 1]
```diff
@@ -18,26 +31,12 @@
...
-import java.io.IOException;
-import java.text.ParseException;
-import java.util.Collection;
-import java.util.Optional;
-import java.util.Set;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
-
-import static com.azure.spring.autoconfigure.aad.Constants.DEFAULT_AUTHORITY_SET;
-import static com.azure.spring.autoconfigure.aad.Constants.ROLE_PREFIX;
```
Comment (backwind1233) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAppRoleStatelessAuthenticationFilter.java:34]: why do this,change format?

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationFilter.java [hunk 2]
```diff
@@ -19,24 +30,12 @@
...
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
-import javax.servlet.http.HttpSession;
-import java.io.IOException;
-import java.net.MalformedURLException;
-import java.text.ParseException;
-import java.util.Optional;
-
-import static com.azure.spring.autoconfigure.aad.Constants.BEARER_PREFIX;
-
```
Comment (backwind1233) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationFilter.java:34]: do we really need change this?

**Review by backwind1233 — COMMENTED** [review]
comments add

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAppRoleStatelessAuthenticationFilter.java (hunk 1 above)
Comment (ZhuXiaoBing-cn) [reply sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAppRoleStatelessAuthenticationFilter.java:34]: It changes due to the automatic import of IDEA.

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationFilter.java (hunk 2 above)
Comment (ZhuXiaoBing-cn) [reply sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationFilter.java:34]: Don't need.

**Comment by chenrujun** [comment]
@ZhuXiaoBing-cn , Maybe we should do like this: https://github.com/spring-projects/spring-security-oauth/blob/2.5.0.RELEASE/spring-security-oauth2/src/main/java/org/springframework/security/oauth2/common/OAuth2AccessToken.java#L21-L28

**Comment by ZhuXiaoBing-cn** [comment]
@chenrujun Okay, I'm going to modify it.

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationFilter.java (hunk 2 above)
Comment (chenrujun) [reply sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationFilter.java:34]: Hi, @backwind1233 , It's OK to keep the diff in the PR. The style is same to other java files in `sdk/spring` folder.

**Review by chenrujun — APPROVED** [review]
LGTM.

**Comment by saragluna** [comment]
We should provide an option to users, letting them include our aad starter for resource server case.

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAppRoleStatelessAuthenticationFilter.java [hunk 3]
```diff
@@ -36,7 +36,11 @@
  * implementation so the principal will not be stored in session. By using roles claim in the token it will not call
  * Microsoft Graph to retrieve users' groups.
  * <p>
- * @deprecated See the <a href="https://github.com/Azure/azure-sdk-for-java/issues/17860">Alternative method</a>.
+ *
+ * @deprecated For AADAppRoleStatelessAuthenticationFilter, suggest use spring-security resource-server directly and
+ * include azure-spring-boot-starter-active-directory dependencies into your project.Automatic configuration of the
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAppRoleStatelessAuthenticationFilter.java:41]: Add space before Automatic

**Review by saragluna — CHANGES_REQUESTED** [review]

**Review by saragluna — APPROVED** [review]
LGTM.

**Pull request merged**